lib/full-site-editing/template-parts.php

```diff
@@ -160,111 +146,6 @@ function set_unique_slug_on_create_template_part( $post_id ) {
-				'description' => __(
-					'General templates often perform a specific role like displaying post content, and are not tied to any particular area.',
-					'gutenberg'
-				),
-				'icon'        => 'layout',
-				'area_tag'    => 'div',
-			),
-			array(
-				'area'        => WP_TEMPLATE_PART_AREA_HEADER,
-				'label'       => __( 'Header', 'gutenberg' ),
-				'description' => __(
-					'The Header template defines a page area that typically contains a title, logo, and main navigation.',
-					'gutenberg'
-				),
-				'icon'        => 'header',
-				'area_tag'    => 'header',
-			),
-			array(
-				'area'        => WP_TEMPLATE_PART_AREA_FOOTER,
-				'label'       => __( 'Footer', 'gutenberg' ),
-				'description' => __(
-					'The Footer template defines a page area that typically contains site credits, social links, or any other combination of blocks.',
-					'gutenberg'
-				),
-				'icon'        => 'footer',
-				'area_tag'    => 'footer',
-			),
-		);
-
-		/**
-		 * Filters the list of allowed template part area values.
-		 *
-		 * @param array $default_areas An array of supported area objects.
-		 */
-		return apply_filters( 'default_wp_template_part_areas', $default_area_definitions );
-	}
-}
-
-/**
- * Checks whether the input 'area' is a supported value.
- * Returns the input if supported, otherwise returns the 'uncategorized' value.
- *
- * @param string $type Template part area name.
- *
- * @return string Input if supported, else the uncategorized value.
- */
-function gutenberg_filter_template_part_area( $type ) {
-	$allowed_areas = array_map(
-		function ( $item ) {
-			return $item['area'];
-		},
-		get_allowed_block_template_part_areas()
-	);
-	if ( in_array( $type, $allowed_areas, true ) ) {
-		return $type;
-	}
-
-	/* translators: %1$s: Template area type, %2$s: the uncategorized template area value. */
-	$warning_message = sprintf( __( '"%1$s" is not a supported wp_template_part area value and has been added as "%2$s".', 'gutenberg' ), $type, WP_TEMPLATE_PART_AREA_UNCATEGORIZED );
-	trigger_error( $warning_message, E_USER_NOTICE );
-	return WP_TEMPLATE_PART_AREA_UNCATEGORIZED;
-}
-
-/**
- * Print a template-part.
- *
- * @param string $part The template-part to print. Use "header" or "footer".
- *
- * @return void
- */
-function gutenberg_block_template_part( $part ) {
-	$template_part = gutenberg_get_block_template( get_stylesheet() . '//' . $part, 'wp_template_part' );
-	if ( ! $template_part || empty( $template_part->content ) ) {
-		return;
-	}
-	echo do_blocks( $template_part->content );
-}
-
-/**
- * Print the header template-part.
- *
- * @return void
- */
-function gutenberg_block_header_area() {
-	gutenberg_block_template_part( 'header' );
-}
 
-/**
- * Print the footer template-part.
- *
- * @return void
- */
-function gutenberg_block_footer_area() {
-	gutenberg_block_template_part( 'footer' );
-}
+remove_action( 'save_post_wp_template_part', 'wp_set_unique_slug_on_create_template_part' );
```

Comment:
If the version in WP core of `wp_set_unique_slug_on_create_template_part` is going to be the same, you can only add a fallback action and bail out early from `set_unique_slug_on_create_template_part`  when the method used in the hook is defined.

By the way, the name of the function prefixes with `gutenberg_` would fit better here.

Reply:
done 😄